Readme.md

```diff
@@ -1,87 +1,153 @@
 # SQLite
 
-Sqlite fork ported to WebAssembly and adapted for the Fluence network. Could be launched and played on the Fluence [dashboard](http://dash.fluence.network/deploy/sqlite).
 
-Based on SQlite version 3.40.1.
+Sqlite fork adapted to work with Fluence protocol. The current version is based on SQlite version 3.40.1. The artifact is sqlite WASM module:
+
+- that is compiled following [Marine runtime ABI conventions](https://fluence.dev/docs/marine-book/marine-rust-sdk/module-abi)
+- to be used with [Marine runtime](https://github.com/fluencelabs/marine)
+
 
 # How to build
 
-This app could be built either with docker
+
+
+Wasm module can be built with either docker-compose
+
+
 
 ```bash
+
 docker-compose up
+
 ```
 
-or by Makefile with [wasi-sdk](https://github.com/WebAssembly/wasi-sdk) installed
+
+
+or using GNU make. There are prerequisites to be installed following this path, namely [wasi-sdk](https://github.com/WebAssembly/wasi-sdk) and marine crate. Take a look at the Dockerfile for the details.
+
 ```bash
+
 make
+
 ```
 
+
+
 # How to use
 
-At now, this fork exports five API functions:
-```cpp
-
-/**
- * Executes given SQL request and returns result in as a pointer to the following structure: | result size (4 bytes, le)| result (size bytes) |.
- *
- * @param sql a pointer to the supplied sql request
- * @param length a size of the supplied sql request
- * @return a pointer to the struct contains result_size and result
- */
-char *invoke(const char *sql, size_t length);
-
-/**
- * Allocates a memory region of a given size. Could be used by Wasm execution environments for byte array passing.
- *
- * @param size a size of allocated memory region
- * @return a pointer to the allocated memory region
- */
-void *allocate(size_t size);
-
-/**
- * Frees a memory region. Could be used by Wasm execution environments for freeing previous memory allocated by `allocate` function.
- *
- * @param ptr a pointer to the previously allocated memory region
- * @param size a size of the previously allocated memory region
- */
-void deallocate(void *ptr, size_t size);
-
-/**
- * Stores one byte by a given address in the module memory.
- *
- * @param ptr a address where byte should be stored
- * @param value a byte to be stored
- */
-void store(void *ptr, char value);
-
-/**
- * Returns one byte by a given address in the module memory.
- *
- * @param ptr a address at which the needed byte is located
- * @return the byte at the given address
- */
-char load(void *ptr);
+The Sqlite Wasm module exports a set of Sqlite C API functions.  There are multiple ways to try these Wasm module functions.
```

Comment:
suggest to write SQLite wherever it's possible

Reply:
What about the following variant:

The SQLite Wasm module exports a set of SQLite C API functions. The easiest way to try is to run it with [Marine](https://github.com/fluencelabs/marine) Wasm runtime and Marine [REPL](https://fluence.dev/docs/marine-book/marine-tooling-reference/marine-repl).

Reply:
Done